bp/src/BPConfiguration.m

```diff
@@ -485,6 +509,29 @@ - (BOOL)loadConfigFile:(NSString *)file withError:(NSError **)errPtr{
     // Pull out two keys that are undocumented but needed for supporting xctest
     self.commandLineArguments = [configDict objectForKey:@"commandLineArguments"];
     self.environmentVariables = [configDict objectForKey:@"environmentVariables"];
+    
+    NSMutableDictionary<NSString *, BPTestPlan*> *tests = [[NSMutableDictionary alloc] init];
+    NSDictionary *testOptions = [configDict objectForKey:@"tests"];
+    for (NSString *key in testOptions) {
+        NSDictionary *planDictionary = [testOptions objectForKey:key];
+        BPTestPlan *plan = [[BPTestPlan alloc] init];
+        plan.testHost = [planDictionary objectForKey:@"test_host"];
+        plan.testHostBundleIdentifier = [planDictionary objectForKey:@"test_host_bundle_identifier"];
+        plan.testBundlePath = [planDictionary objectForKey:@"test_bundle_path"];
+        plan.uiTargetAppPath = [planDictionary objectForKey:@"ui_target_app_path"];
+        plan.environment = [planDictionary objectForKey:@"environment"];
+        plan.arguments = [planDictionary objectForKey:@"arguments"];
+        
+        if (!plan.isValid) {
+            BP_SET_ERROR(errPtr, @"Invalid BPTestPlan configuration: %@", plan);
```

Comment:
Can we explain to the user what is missing? I like errors that tell you what to do... like "Add this to your config to fix it"